### PR TITLE
Adding shellcheck GitHub Action

### PR DESCRIPTION
**Pull request by Lucas-C**
Content of this PR:

* it introduces a GitHub Action to perform [`shellcheck` static code analysis](https://github.com/koalaman/shellcheck) of the `.sh` files in `GitLab/scripts/`. This check could be extended to analyze more `.sh` files of this repo later on.
* it includes fixes for all `shellcheck` warnings on the scripts in `Gitlab/scripts/`
* it adds a check that authentication on `$ORKA_ENDPOINT/token` is successful in `prepare.sh`
* it sets a `ConnectTimeout` of 60 seconds when performing the first SSH connection in `prepare.sh`

### CODE REVIEW

**Comment by Lucas-C** [comment]
GitHup Actions pipeline result: https://github.com/voyages-sncf-technologies/orka-integrations/actions/runs/931500392

File: GitLab/scripts/base.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 BUILD_ID="runner-$CUSTOM_ENV_CI_RUNNER_ID-project-$CUSTOM_ENV_CI_PROJECT_ID-concurrent-$CUSTOM_ENV_CI_CONCURRENT_PROJECT_ID"
-CONNECTION_INFO_ID=$BUILD_ID-connection-info
+export CONNECTION_INFO_ID=$BUILD_ID-connection-info
```
Comment (ispasov) [inline GitLab/scripts/base.sh:4]: Is there a need for this export?

File: GitLab/scripts/cleanup.sh [hunk 2]
```diff
@@ -1,26 +1,28 @@
 #!/bin/bash
 
+set -o errexit -o nounset -o pipefail
```
Comment (ispasov) [inline GitLab/scripts/cleanup.sh:3]: I would keep the short syntax.
It is consistent with the rest of the repository. And there is really no reason for the long version.
Same goes for all other usages of `set -eu`

**Comment by ispasov** [comment]
Thanks for the contribution. I have a few minor comments.

File: GitLab/scripts/base.sh (hunk 1 above)
Comment (Lucas-C) [reply GitLab/scripts/base.sh:4]: The need is to comply with this `shellcheck` rule: https://github.com/koalaman/shellcheck/wiki/SC2034
We could also disable it

File: GitLab/scripts/cleanup.sh (hunk 2 above)
Comment (Lucas-C) [reply GitLab/scripts/cleanup.sh:3]: I try to follow a set of best practices regarding shell scripting, inspired by such articles:

* https://kvz.io/bash-best-practices.html
> Use long options (logger --priority vs logger -p). If you're on cli, abbreviations make sense for efficiency. but when you're writing reusable scripts a few extra keystrokes will pay off in readability and avoid ventures into man pages in the future by you or your collaborators.

* https://www.franzoni.eu/shell-scripting-when-to-use-the-long-or-the-shor/
> But whenever you write a script that should, later on, be used and read by other people - or just yourself in the future - the long form is often self-explanatory.

Of course if you really want to, I'll switch to short options

**Comment by Lucas-C** [comment]
Comments answered.
Thanks for the review!

File: GitLab/scripts/base.sh (hunk 1 above)
Comment (ispasov) [reply GitLab/scripts/base.sh:4]: Got it.
Well, I am ok with both.

File: GitLab/scripts/cleanup.sh (hunk 2 above)
Comment (ispasov) [reply GitLab/scripts/cleanup.sh:3]: I personally find those options in the `if the option is very common` category from the second link.
If you are familiar with them you would recognise them in both short and long format. If you are not, then even if you see the long format, you most probably need to check the manpage.
So I vote for the short syntax here.

**Review by ispasov — CHANGES_REQUESTED** [review]
I think we are good to go.
I would just change the `set` options to use the short syntax

File: GitLab/scripts/cleanup.sh (hunk 2 above)
Comment (Lucas-C) [reply GitLab/scripts/cleanup.sh:3]: As you wish. I have just made the change.

**Comment by Lucas-C** [comment]
Change made. I think this is ready to be merged

**Comment by ispasov** [comment]
There are still two occurrences of the long syntax - in `run.sh` and `prepare.sh`. Can you check them out, please?

**Comment by Lucas-C** [comment]
Done

**Review by ispasov — APPROVED** [review]

**Pull request merged**